### PR TITLE
Add the rest api section

### PR DESCRIPTION
**Pull request by CharlleDaniel**
this pr is able to:
 - add the section rest_api.

### CODE REVIEW

**Comment by skovic** [comment]
@CharlleDaniel I think we want this in a separate document. Could you move your files to the _physical_infra_provider_rest_api_ directory (you can create a _rest_api/topics_ subdirectory in it for your files) and update its main.adoc file (a template is already there) to point to your files? Thanks

**Comment by CharlleDaniel** [comment]
Hi @skovic, thanks for your review. I agree with you and I will work on this. :+1:

**Comment by skovic** [comment]
@CharlleDaniel Also Red Hat has suggested that we use asciidoc style lists instead of explicitly numbering them. For example, ". Item A   . Item B" will yield "1. Item A   2. Item B". So please consider updating the steps of your procedures. You can see examples here: http://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#lists

File: physical_infra_provider_rest_api/usage/topics/actions_physical_infra_providers.adoc [hunk 1]
```diff
@@ -0,0 +1,58 @@
+=== Actions of the a Physical Infra Provider Lenovo
```
Comment (skovic) [inline physical_infra_provider_rest_api/usage/topics/actions_physical_infra_providers.adoc]: Maybe change this to "Physical Infrastructure Provider Actions"

File: physical_infra_provider_rest_api/usage/topics/actions_physical_servers.adoc [hunk 2]
```diff
@@ -0,0 +1,61 @@
+=== Actions of the a Physical Server
```
Comment (skovic) [inline physical_infra_provider_rest_api/usage/topics/actions_physical_servers.adoc]: Maybe change this to "Physical Server Actions"

File: physical_infra_provider_rest_api/usage/topics/index.adoc [hunk 3]
```diff
@@ -0,0 +1,3 @@
+== REST API
+
+The Rest API can be used for get information about providers, physical infra providers, physical servers and so on, you should use the CloudForms Rest API if you need data information about you machines in a personal application. The Rest API is able to of return a data information very significant or just a custom data information, try it.
```
Comment (skovic) [inline physical_infra_provider_rest_api/usage/topics/index.adoc]: Maybe change this to "The REST API can be used for retrieving information about providers, physical infrastructure providers, and physical servers. You can use the CloudForms REST API if you need information about the servers in your physical infrastructure."

File: physical_infra_provider_rest_api/usage/topics/get_physical_servers.adoc [hunk 4]
```diff
@@ -0,0 +1,27 @@
+=== Get Physical Servers
+
+. Log in to CloudForms Management Engine API as a user who has permission to access the API. The default user is admin, password smartvm.
+
+. Use the following url structure for access the api **"<CLOUDFORMS ADDRESS>:<PORT>/api/" (ex. localhost:3000/api/)**.
+
+. Use the following request for get all physical servers. **(ex. localhost:3000/api/physical_servers)**.
+
+. It’s returned a JSON with the following keys**
```
Comment (skovic) [inline physical_infra_provider_rest_api/usage/topics/get_physical_servers.adoc]: Maybe you can put the keys into a table like you did in step 4 of section 1.5 (Get a Physical Server by ID).

File: physical_infra_provider_rest_api/usage/topics/get_providers.adoc [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+. Log in to CloudForms Management Engine API as a user who has permission to access the API. The default user is admin, password smartvm.
+
+. Use the following url structure for access the api **"<CLOUDFORMS ADDRESS>:<PORT>/api/" (ex. localhost:3000/api/)**.
+
+. Use the following request for get all provider’s resource. **(ex. localhost:3000/api/providers)**.
+
+. It’s returned a JSON with the following keys
+    
+    .. name **(URI name)**.
```
Comment (skovic) [inline physical_infra_provider_rest_api/usage/topics/get_providers.adoc]: Maybe you can put the keys into a table like you did in step 4 of section 1.5 (Get a Physical Server by ID).

File: physical_infra_provider_rest_api/usage/topics/get_physical_server_by_id.adoc [hunk 6]
```diff
@@ -0,0 +1,34 @@
+=== Get a Physical Server by ID
```
Comment (skovic) [inline physical_infra_provider_rest_api/usage/topics/get_physical_server_by_id.adoc]: Maybe change this to "Retrieving Physical Server Details by ID"

File: physical_infra_provider_rest_api/usage/topics/get_physical_servers.adoc [hunk 7]
```diff
@@ -0,0 +1,27 @@
+=== Get Physical Servers
```
Comment (skovic) [inline physical_infra_provider_rest_api/usage/topics/get_physical_servers.adoc]: Maybe change this to "Retrieving Physical Server Details"

File: physical_infra_provider_rest_api/usage/topics/get_provider_by_id.adoc [hunk 8]
```diff
@@ -0,0 +1,27 @@
+=== Get a Provider by ID
```
Comment (skovic) [inline physical_infra_provider_rest_api/usage/topics/get_provider_by_id.adoc]: Maybe change this to "Retrieving Provider Details by ID"

File: physical_infra_provider_rest_api/usage/topics/get_providers.adoc [hunk 9]
```diff
@@ -0,0 +1,27 @@
+=== Get Providers
```
Comment (skovic) [inline physical_infra_provider_rest_api/usage/topics/get_providers.adoc]: Maybe change this to "Retrieving Provider Details"

**Comment by skovic** [comment]
@CharlleDaniel Do you think you can get these changes in today? Thanks

**Comment by skovic** [comment]
Merging the PR as is. The changes will be made in another PR.

**Pull request merged**